src/autosuggest/__tests__/input-controller.test.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useKeyboardHandler } from '../controller';
+import { renderHook } from '../../__tests__/render-hook';
+import { CancelableEventHandler, BaseKeyDetail, fireCancelableEvent } from '../../internal/events';
+import { KeyCode } from '../../internal/keycode';
+
+describe('useKeyboardHandler', () => {
+  const open = true;
+  const onPressArrowDown = jest.fn();
+  const onPressArrowUp = jest.fn();
+  const onPressEnter = jest.fn();
+  const onKeyDown = jest.fn();
+  const keyDetail = {
+    key: '',
+    ctrlKey: false,
+    shiftKey: false,
+    altKey: false,
+    metaKey: false,
+  };
+  let handleKeyDown: CancelableEventHandler<BaseKeyDetail>;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    const { result } = renderHook(useKeyboardHandler, {
+      initialProps: { open, onPressArrowDown, onPressArrowUp, onPressEnter, onKeyDown },
+    });
+    handleKeyDown = result.current;
+  });
+
+  test('triggers respective key handlers', () => {
+    fireCancelableEvent(handleKeyDown, { keyCode: KeyCode.down, ...keyDetail });
+    expect(onPressArrowDown).toBeCalled();
+    fireCancelableEvent(handleKeyDown, { keyCode: KeyCode.up, ...keyDetail });
+    expect(onPressArrowUp).toBeCalled();
+    fireCancelableEvent(handleKeyDown, { keyCode: KeyCode.enter, ...keyDetail });
+    expect(onPressEnter).toBeCalled();
+  });
+
+  test('does not proxy "arrowDown" and "arrowUp" key downs to customer`s handler', () => {
+    fireCancelableEvent(handleKeyDown, { keyCode: KeyCode.up, ...keyDetail });
+    fireCancelableEvent(handleKeyDown, { keyCode: KeyCode.down, ...keyDetail });
+    expect(onKeyDown).not.toBeCalled();
+  });
+
+  test('proxies every other keys to the customer`s handler', () => {
```

Comment:
typo: "every other keys" should be "every other key"

(also, in "customer`s" there's a backtick where an apostrophe (') should be)